src/test/misc/map.spec.ts

```diff
@@ -1,25 +1,144 @@
 import * as chai from 'chai';
 import * as chaiAsPromised from 'chai-as-promised';
 
-import { map, sleep } from '../../lib/misc';
+import { flatMapAsync, map, mapAsync, mapObjectAsync, sleep } from '../../lib/misc';
 
 chai.use(chaiAsPromised as any);
 const assert = chai.assert;
 
 describe('> map', () => {
 
     it('> should return values in correct order', async () => {
-        assert.deepEqual(await map([2, 5], async (val: number) => {
+        const arr: number[] = [];
+        assert.deepEqual(await map([2, 5], async val => {
+            if (val === 2) await sleep(10);
```

Comment:
Avec un npm ci à la racine, j'ai des erreur de compilation ! (node 18, mais ça devrait pas impacter)

```
src/test/misc/map.spec.ts:69:17 - error TS2367: This condition will always return 'false' since the types 'Promise<number>' and '2' have no overlap.

69             if (val === 2) await sleep(10);
                   ~~~~~~~~~

  src/test/misc/map.spec.ts:69:17
    69             if (val === 2) await sleep(10);
                       ~~~~~~~~~
    Did you forget to use 'await'?

src/test/misc/map.spec.ts:70:22 - error TS2345: Argument of type 'Promise<number>' is not assignable to parameter of type 'number'.

70             arr.push(val);
                        ~~~

  src/test/misc/map.spec.ts:70:22
    70             arr.push(val);
                            ~~~
    Did you forget to use 'await'?

src/test/misc/map.spec.ts:71:20 - error TS2362: The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.

71             return val * index;
                      ~~~

  src/test/misc/map.spec.ts:71:20
    71             return val * index;
                          ~~~
    Did you forget to use 'await'?

src/test/misc/map.spec.ts:116:17 - error TS7053: Element implicitly has an 'any' type because expression of type '0' can't be used to index type 'Promise<number[]>'.
  Property '0' does not exist on type 'Promise<number[]>'.

116             if (val[0] === 2) await sleep(10);
                    ~~~~~~
```

Reply:
fixed